app/views/UnderpaymentTypeView.scala.html

```diff
@@ -0,0 +1,83 @@
+@*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@this(
+        layout: templates.Layout,
+        formHelper: FormWithCSRF,
+        errorSummary: components.errorSummary,
+        button: components.button,
+        govukCheckboxes: GovukCheckboxes
+)
+
+@(form: Form[UnderpaymentType], data: UnderpaymentType)(implicit request: Request[_], messages: Messages)
+
+@layout(
+    pageTitle = Some(messages("underpaymentType.pageTitle")),
+    form = Some(form),
+    customBackLinkUrl = Some(controllers.routes.UnderpaymentTypeController.onLoad)
+) {
+
+    @errorSummary(form.errors)
+
+    @formHelper(action = controllers.routes.UnderpaymentTypeController.onSubmit()) {
+
+        @govukCheckboxes(Checkboxes(
```

Comment:
This should be a separate checkboxes component. If you pass in the form then it will contain whether any item have been previously selected.

In the same way NumberOfEntries creates the list of RadioButtons in it's model, the same could be done for the checkboxItems in UnderpaymentType.